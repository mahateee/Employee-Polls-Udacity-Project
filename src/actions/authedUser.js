export const SET_AUTHED_USER = "SET_AUTHED_USER ";
export function setAuthedUser(authedUser) {
  return {
    type: SET_AUTHED_USER,
    authedUser,
  };
}

export function handleLogin(username, password) {
  return (dispatch, getState) => {
    const { users } = getState();

    const user = Object.values(users).find(
      (user) => user.id === username && user.password === password
    );

    if (!!user) {
      return dispatch(setAuthedUser(user));
    }
  };
}
