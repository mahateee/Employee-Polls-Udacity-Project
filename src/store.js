import { legacy_createStore as createStore } from "redux";
import { rootReducer } from "./reducers/index";
import middleware from "./middleware";

export const store = createStore(rootReducer, middleware);
