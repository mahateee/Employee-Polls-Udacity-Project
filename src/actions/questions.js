import { _saveQuestion } from "../utils/_DATA";
import { saveQuestionAnswer, saveQuestion } from "../utils/api";
import { addAnswerToUser, addQuestionToUser } from "./users";
export const ADD_QUESTION = "ADD_QUESTION";
export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const ADD_ANSWER_TO_QUESTION = "ADD_ANSWER_TO_QUESTION";
export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}
export function addAnswerToQuestion({ authedUser, qid, answer }) {
  return {
    type: ADD_ANSWER_TO_QUESTION,
    authedUser,
    qid,
    answer,
  };
}

function addQuestion(question) {
  return {
    type: ADD_QUESTION,
    question,
  };
}
export function handleAddAnswer(info) {
  return (dispatch) => {
    saveQuestionAnswer(info)
      .then(() => {
        dispatch(addAnswerToQuestion(info));
        dispatch(addAnswerToUser(info));
      })
      .catch((e) => {
        console.warn("Error in handleToggleTweet: ", e);
        dispatch(addAnswerToQuestion(info));
        dispatch(addAnswerToUser(info));

        alert("There was an error liking the tweet. Try again.");
      });
  };
}
export function handleSaveQuestion(info) {
  return (dispatch) => {
    return saveQuestion(info).then((info) => {
      dispatch(addQuestion(info));
      dispatch(addQuestionToUser(info));
    });
  };
}
