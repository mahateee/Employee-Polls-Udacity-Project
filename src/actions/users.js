import { saveQuestionAnswer } from "../utils/api";
import { addAnswerToQuestion } from "./questions";
export const RECEIVE_USERS = "RECEIVE_USERS";
export const ADD_ANSWER_TO_USER = "ADD_ANSWER_TO_USER";
export const ADD_USER_QUESTION = "ADD_QUESTION_TO_USER";
export function receiveUsers(users) {
  return {
    type: RECEIVE_USERS,
    users,
  };
}

export function addAnswerToUser({ authedUser, qid, answer }) {
  return {
    type: ADD_ANSWER_TO_USER,
    authedUser,
    qid,
    answer,
  };
}
export function addQuestionToUser({ author, id }) {
  return {
    type: ADD_USER_QUESTION,
    author,
    qid: id,
  };
}
