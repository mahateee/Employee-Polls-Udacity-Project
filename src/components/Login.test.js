import { render, screen } from "@testing-library/react";
import * as React from "react";
import Login from "./Login";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import { setAuthedUser } from "../actions/authedUser";
describe("Login", () => {
  it("should render the component", () => {
    // store.dispatch(setAuthedUser("sarahedo"));

    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });
  it("will have all expected fields", () => {
    // store.dispatch(setAuthedUser("sarahedo"));

    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    var username = screen.getByTestId("username-input");
    var password = screen.getByTestId("password-input");
    expect(username).toBeInTheDocument();
    expect(password).toBeInTheDocument();

    var submitButton = screen.getByText("Sign in");
    expect(submitButton).toBeInTheDocument();
  });
});
