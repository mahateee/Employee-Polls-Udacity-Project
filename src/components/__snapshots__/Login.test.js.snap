// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="bg-gray-50 dark:bg-gray-900"
      >
        <div
          class="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0"
        >
          <a
            class="flex items-center mb-6 text-2xl font-semibold text-gray-900 dark:text-white"
            href="#"
          >
            <img
              alt="logo"
              class="w-8 h-8 mr-2"
              src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/logo.svg"
            />
            Flowbite
          </a>
          <div
            class="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
          >
            <div
              class="p-6 space-y-4 md:space-y-6 sm:p-8"
            >
              <h1
                class="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
              >
                Sign in to your account
              </h1>
              <form
                action="#"
                class="space-y-4 md:space-y-6"
              >
                <div>
                  <label
                    class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                    for="usename"
                  >
                    Your username
                  </label>
                  <input
                    class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                    data-testid="username-input"
                    id="username"
                    name="username"
                    placeholder="name"
                    type="username"
                    value="sarahedo"
                  />
                </div>
                <div>
                  <label
                    class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                    data-testid="password-input"
                    id="password"
                    name="password"
                    placeholder="••••••••"
                    type="password"
                    value="password123"
                  />
                </div>
                <button
                  class="w-full text-white bg-blue-600 hover:bg-primary-700 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
                  type="submit"
                >
                  Sign in
                </button>
              </form>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="bg-gray-50 dark:bg-gray-900"
    >
      <div
        class="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0"
      >
        <a
          class="flex items-center mb-6 text-2xl font-semibold text-gray-900 dark:text-white"
          href="#"
        >
          <img
            alt="logo"
            class="w-8 h-8 mr-2"
            src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/logo.svg"
          />
          Flowbite
        </a>
        <div
          class="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
        >
          <div
            class="p-6 space-y-4 md:space-y-6 sm:p-8"
          >
            <h1
              class="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
            >
              Sign in to your account
            </h1>
            <form
              action="#"
              class="space-y-4 md:space-y-6"
            >
              <div>
                <label
                  class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                  for="usename"
                >
                  Your username
                </label>
                <input
                  class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  data-testid="username-input"
                  id="username"
                  name="username"
                  placeholder="name"
                  type="username"
                  value="sarahedo"
                />
              </div>
              <div>
                <label
                  class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                  for="password"
                >
                  Password
                </label>
                <input
                  class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  data-testid="password-input"
                  id="password"
                  name="password"
                  placeholder="••••••••"
                  type="password"
                  value="password123"
                />
              </div>
              <button
                class="w-full text-white bg-blue-600 hover:bg-primary-700 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
                type="submit"
              >
                Sign in
              </button>
            </form>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
